src/App.jsx

```diff
@@ -1,15 +1,64 @@
 import { hot } from 'react-hot-loader';
-import React from 'react';
+import React, { useState } from 'react';
+import { Authorize } from 'react-auth0-components';
 import { ThemeProvider } from '@material-ui/styles';
+import AuthContext from './utils/AuthContext';
+import { USER_SESSION } from './utils/constants';
 import theme from './theme';
 import Main from './Main';
 
-const App = () => (
-  <div>
-    <ThemeProvider theme={theme}>
-      <Main />
-    </ThemeProvider>
-  </div>
-);
+const App = () => {
+  const [authorize, setAuthorize] = useState(false);
+  const [authContext, setAuthContext] = useState({
+    authorize: () => setAuthorize(true),
+    unauthorize: () => {
+      setAuthorize(false);
+      setAuthContext({
+        ...authContext,
+        user: null,
+      });
+    },
+    user: null,
+  });
+
+  return (
+    <AuthContext.Provider value={authContext}>
+      <ThemeProvider theme={theme}>
+        <Authorize
+          authorize={authorize}
+          popup
+          domain={process.env.AUTH0_DOMAIN}
+          clientID={process.env.AUTH0_CLIENT_ID}
+          audience={process.env.AUTH0_AUDIENCE}
+          redirectUri={process.env.AUTH0_REDIRECT_URI}
+          responseType={process.env.AUTH0_RESPONSE_TYPE}
+          scope={process.env.AUTH0_SCOPE}
+          render={() => {
+            const session = localStorage.getItem(USER_SESSION);
+
+            if (session) {
+              const user = JSON.parse(session);
+              const expires = new Date(user.expiration);
```

Comment:
Is this supposed to handle refresh tokens? If so, I'm not certain it's working. I tried setting `expires` to be slightly in the future, and AFAICT it never tried to refresh.

Reply:
> Is this supposed to handle refresh tokens?

It should but I don't think it was working before. I pushed a fix. I'll test it tomorrow.

Reply:
The timer for renewal uses the expiration token in localStorage. This is done in https://github.com/mozilla-frontend-infra/react-auth0-components/blob/99c32611abd9bc5ada5ca5daa8a944d08fa055d3/src/components/Authorize/index.jsx#L264. I don't think setting `expires` here will affect the renewal timer.